Compiler/src/ssair/passes.jl

```diff
@@ -1511,6 +1511,10 @@ function sroa_pass!(ir::IRCode, inlining::Union{Nothing,InliningState}=nothing)
             used_ssas[x.id] -= 1
         end
         ir = complete(compact)
+        # remove any use that has been optimized away by the DCE
+        for (intermediaries, defuse) in values(defuses)
+            filter!(x -> ir[SSAValue(x.idx)][:stmt] !== nothing, defuse.uses)
+        end
```

Comment:
Since we don't use `defuses` until here, we could also defer its computation to after the DCE, which would probably be more robust. I just went with the simplest fix here.

I'm also assuming in this update that definitions (`setfield!`) won't be optimized away by DCE, if that is incorrect I can also update these.

Reply:
No, `setfield!` won't be DCEed here.